cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/PreviewRunnerTwillApplication.java

```diff
@@ -32,24 +34,37 @@
   private final URI cConfFileURI;
   private final URI hConfFileURI;
   private final ResourceSpecification resourceSpec;
+  private final Optional<ResourceSpecification> artifactLocalizerResourceSpec;
 
-  public PreviewRunnerTwillApplication(URI cConfFileURI, URI hConfFileURI, ResourceSpecification resourceSpec) {
+  public PreviewRunnerTwillApplication(URI cConfFileURI, URI hConfFileURI, ResourceSpecification resourceSpec,
+                                       Optional<ResourceSpecification> artifactLocalizerResourceSpec) {
     this.cConfFileURI = cConfFileURI;
     this.hConfFileURI = hConfFileURI;
     this.resourceSpec = resourceSpec;
+    this.artifactLocalizerResourceSpec = artifactLocalizerResourceSpec;
   }
 
   @Override
   public TwillSpecification configure() {
-    return TwillSpecification.Builder.with()
+    TwillSpecification.Builder.MoreRunnable runnables = TwillSpecification.Builder.with()
       .setName(NAME)
-      .withRunnable()
-        .add(new PreviewRunnerTwillRunnable("cConf.xml", "hConf.xml"), resourceSpec)
+      .withRunnable();
+
+    TwillSpecification.Builder.RunnableSetter runnableSetter =
+    runnables.add(new PreviewRunnerTwillRunnable("cConf.xml", "hConf.xml"), resourceSpec)
       .withLocalFiles()
         .add("cConf.xml", cConfFileURI)
         .add("hConf.xml", hConfFileURI)
-      .apply()
-      .anyOrder()
-      .build();
+      .apply();
+
+    if (artifactLocalizerResourceSpec.isPresent()) {
```

Comment:
nit: use ifPresent (and consumer) to avoid if block.

Reply:
Done